solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,522 @@
+
+        uint256[] memory networkTokenConversionAmounts = new uint256[](tokens.length);
+        uint256 grossNetworkTokenConversionAmount = 0;
+        uint256 totalGovTokenAmountToBurn = 0;
+
+        for (uint256 i = 0; i < paths.length; ++i) {
+            // avoid empty conversions
+            uint256 amount = amounts[i];
+            if (amount == 0) {
+                continue;
+            }
+
+            address[] memory path = paths[i];
+            IERC20 token = IERC20(path[0]);
+            uint256 value = 0;
+
+            if (token == _govToken) {
+                // if the source token is the governance token, don't try to convert it either, but rather include it in
+                // the amount to burn
+                //
+                // Please note that networkTokenConversionAmounts[i] will remain 0 in this case
+                totalGovTokenAmountToBurn = totalGovTokenAmountToBurn.add(amount);
+            } else if (token == _networkToken) {
+                // if the source token is the network token, don't try to convert it, but rather add its amount to the
+                // total amount to convert to the governance token
+                grossNetworkTokenConversionAmount = grossNetworkTokenConversionAmount.add(amount);
+
+                networkTokenConversionAmounts[i] = amount;
+            } else {
+                if (token != NATIVE_TOKEN_ADDRESS) {
+                    // if the source token is a regular token, approve the converter to withdraw the token amount
+                    ensureAllowance(token, network, amount);
+                } else {
+                    // if the source token is actually an ETH reserve, make sure to pass its value to the network
+                    value = amount;
+                }
+
+                // perform the actual conversion and optionally send ETH to the network
+                uint256 networkTokenConversionAmount =
+                    network.convertByPath2{ value: value }(path, amount, 1, address(this));
+
+                // update network conversion amounts
+                networkTokenConversionAmounts[i] = networkTokenConversionAmount;
+                grossNetworkTokenConversionAmount = grossNetworkTokenConversionAmount.add(networkTokenConversionAmount);
+            }
+        }
+
+        // calculate the burn incentive fee and reduce it from the total amount to convert
+        (uint256 netNetworkTokenConversionAmount, uint256 incentiveFeeAmount) =
+            applyIncentiveFee(grossNetworkTokenConversionAmount);
+
+        // approve the governance token converter to withdraw the network token amount
+        ensureAllowance(_networkToken, network, netNetworkTokenConversionAmount);
+
+        // convert all network token amounts to the governance token
+        totalGovTokenAmountToBurn = totalGovTokenAmountToBurn.add(
+            network.convertByPath2(govPath, netNetworkTokenConversionAmount, 1, address(this))
+        );
```

Comment:
Same question about min-return, as above. Perhaps we should first call `rateByPath` in order to determine the min-return (e.g., 90% of the returned value)?

Reply:
See my comment above